tt.go

```diff
@@ -0,0 +1,278 @@
+}
+
+// GetDefaultStringToTTMapperOpts returns default string to tt mappers options.
+func GetDefaultStringToTTMapperOpts() StringToTTMapperOpts {
+	return StringToTTMapperOpts{
+		ThSeparators:  defaultThSeparators,
+		DecSeparators: defaultDecSeparators,
+		NullValue:     defaultNullValue,
+	}
+}
+
+// StringToTTMapperFactory is the default TypeToTTMapperFactory for strings.
+// To customize the creation of mappers, inherit from it and override the necessary methods.
+type StringToTTMapperFactory struct {
+	opts StringToTTMapperOpts
+}
+
+// NewStringToTTMapperFactory creates StringToTTMapperFactory.
+func NewStringToTTMapperFactory() *StringToTTMapperFactory {
+	return &StringToTTMapperFactory{
+		opts: StringToTTMapperOpts{
+			ThSeparators:  defaultThSeparators,
+			DecSeparators: defaultDecSeparators,
+			NullValue:     defaultNullValue,
+		},
+	}
+}
+
+func (*StringToTTMapperFactory) MakeTypeToBooleanMapper() Mapper[string, any] {
+	return MakeStringToBoolMapper()
+}
+
+func (*StringToTTMapperFactory) MakeTypeToStringMapper() Mapper[string, any] {
+	return MakeIdentityMapper[string]()
+}
+
+func (fac *StringToTTMapperFactory) MakeTypeToUnsignedMapper() Mapper[string, any] {
+	return MakeStringToUIntMapper(fac.opts.ThSeparators)
+}
+
+func (*StringToTTMapperFactory) MakeTypeToDatetimeMapper() Mapper[string, any] {
+	return MakeStringToDateTimeMapper()
+}
+
+func (*StringToTTMapperFactory) MakeTypeToUUIDMapper() Mapper[string, any] {
+	return MakeStringToUUIDMapper()
+}
+
+func (*StringToTTMapperFactory) MakeTypeToMapMapper() Mapper[string, any] {
+	return MakeStringToMapMapper()
+}
+
+func (*StringToTTMapperFactory) MakeTypeToArrayMapper() Mapper[string, any] {
+	return MakeStringToSliceMapper()
+}
+
+func (*StringToTTMapperFactory) MakeTypeToVarbinaryMapper() Mapper[string, any] {
+	return MakeStringToBinaryMapper()
+}
+
+func (fac *StringToTTMapperFactory) MakeTypeToDoubleMapper() Mapper[string, any] {
+	return MakeStringToFloatMapper(fac.opts.ThSeparators, fac.opts.DecSeparators)
+}
+
+func (fac *StringToTTMapperFactory) MakeTypeToDecimalMapper() Mapper[string, any] {
+	return MakeStringToDecimalMapper(fac.opts.ThSeparators, fac.opts.DecSeparators)
+}
+
+func (fac *StringToTTMapperFactory) MakeTypeToIntegerMapper() Mapper[string, any] {
+	return MakeSequenceMapper([]Mapper[string, any]{
+		MakeStringToUIntMapper(fac.opts.ThSeparators),
+		MakeStringToIntMapper(fac.opts.ThSeparators),
+	})
+}
+
+func (fac *StringToTTMapperFactory) MakeTypeToNumberMapper() Mapper[string, any] {
+	return MakeSequenceMapper([]Mapper[string, any]{
+		MakeStringToUIntMapper(fac.opts.ThSeparators),
+		MakeStringToIntMapper(fac.opts.ThSeparators),
+		MakeStringToFloatMapper(fac.opts.ThSeparators, fac.opts.DecSeparators),
+	})
+}
+
+func (fac *StringToTTMapperFactory) MakeTypeToAnyMapper() Mapper[string, any] {
+	return MakeSequenceMapper([]Mapper[string, any]{
+		MakeStringToUIntMapper(fac.opts.ThSeparators),
+		MakeStringToIntMapper(fac.opts.ThSeparators),
```

Comment:
It must call it's own implementation, not a function. No options must be provided here.